src/oData/jsonToOData.ts

```diff
@@ -64,6 +74,26 @@ const handlePrimitiveFilter = (
 		if (regexp.flags != null && regexp.flags !== 'i') {
 			throw new Error(`Regex flag ${regexp.flags} is not supported`);
 		}
+		if (
+			value.$comment === 'starts_with' ||
+			value.$comment === 'not_starts_with'
+		) {
```

Comment:
Are these necessary in this PR? If no, then let's drop them to keep it cleaner.

Reply:
that was not intended